### DISCUSSION TITLE
Different versions of Ruffle

### DISCUSSION THREAD

**Discussion by Mark0669** [ruffle-rs/ruffle#9921] [Ideas]
I've had a year old web hosted version of Ruffle installed on a site running AS1 swfs - it works brilliantly! Yesterday I installed the latest version of Ruffle, again all works great but there was a noticeable increase in load times. As Ruffle develops to accommodate  AS2 and AS3 I suspect will probably increase further. My question therefore is would it be possible to create a version that solely support AS1 given most code is supported and work flawlessly.

Cheers,
Mark

**Comment by n0samu** [comment]
This is a great idea, and one we've been discussing for some time. We'd like to allow website owners to customize exactly what features are included in the Ruffle builds they deploy, to allow optimization of load times and download sizes. For now, our AVM1 and AVM2 implementations are not cleanly delineated, but hopefully we can change that in the future.

As for the long startup time, I believe is mostly caused by the additional rendering features that have been implemented recently, specifically blend modes. Each new set of features means an additional rendering 'pipeline' needs to be initialized before Ruffle starts, on the off-chance that the Flash content will use it. Maybe we can provide an option to disable blend modes and other expensive features to reduce initialization time. Another option, strangely enough, would be to compile Ruffle with a rendering backend that does not include an implementation of blend modes, because that lack of functionality reduces the startup time. If [a feature I just implemented](https://github.com/ruffle-rs/ruffle/pull/9904) would be accepted, it would be possible to choose the rendering backend without even recompiling Ruffle. But judging from the comments of the core developers, that is unlikely to happen.

**Reply by adrian17** [reply]
For the record, indeed AS2 contributes very small, almost immeasurably, to the load times; most of time increase is indeed caused by the new renderer initialization (and Ruffle binary download time - its size also increased a lot with the new renderer - but that's unavoidable without recompiling Ruffle to disable it).

**Reply by n0samu** [reply]
My feature has been added, so you can now choose a rendering backend without recompiling Ruffle using the `preferredRenderer` option.  [See our wiki for more information](https://github.com/ruffle-rs/ruffle/wiki/Using-Ruffle#configuration-options).

**Comment by Lena567Lena** [comment]
Different versions of Ruffle would be great
